mv3/addon/background.ts

```diff
@@ -0,0 +1,92 @@
+import browser, { Runtime } from "webextension-polyfill";
+
+import { create } from "ipfs-http-client";
+import IsIpfs from "is-ipfs";
+const check = String.fromCodePoint(0x2714);
+
+let lifeline: Runtime.Port;
+const client = create({ url: "http://localhost:5001/api/v0" });
+
+browser.runtime.onConnect.addListener((port: Runtime.Port) => {
+  if (port.name === "keepAlive") {
+    lifeline = port;
+    setTimeout(keepAliveForced, 295e3); // 5 minutes minus 5 seconds
+    port.onDisconnect.addListener(keepAliveForced);
+  }
+});
+
+function keepAliveForced() {
+  if (lifeline) lifeline.disconnect();
+  lifeline = null;
+  keepAlive();
+}
+
+async function keepAlive() {
+  if (lifeline) return;
+  for (const tab of await browser.tabs.query({ url: "*://*/*" })) {
+    try {
+      await browser.scripting.executeScript({
+        target: { tabId: tab.id },
+        func: () => browser.runtime.connect({ name: "keepAlive" }),
+      });
+      browser.tabs.onUpdated.removeListener(retryOnTabUpdate);
+      return;
+    } catch (e) {
+      console.error(e);
+    }
+  }
+  browser.tabs.onUpdated.addListener(retryOnTabUpdate);
+}
+
+function isNewTabURL(url) {
+  if (!url) return true;
+  return url.includes("chrome://");
+}
+
+async function resolveDNSAndAddRule(domain) {
+  const cid = await client.dns(domain);
+  if (!cid) return;
+
+  const id = Math.floor(Math.random() * 29999);
+  // @ts-ignore
```

Comment:
`@ts-ignore` like this triggers github action check which prints same warning 3 times below (can we remove this noise?)